components/eventing-controller/pkg/handlers/utils.go

```diff
@@ -43,6 +44,49 @@ type MessagingBackend interface {
 
 const charset = "abcdefghijklmnopqrstuvwxyz0123456789"
 
+// NameMapper is used to map Kyma-specific resource names to their corresponding name on other
+// (external) systems, e.g. on different eventing backends, the same Kyma subscription name
+// could map to a different name.
+type NameMapper interface {
+	MapSubscriptionName(sub *eventingv1alpha1.Subscription) string
+}
+
+// bebSubscriptionNameMapper maps a Kyma subscription to an ID that can be used on the BEB backend,
+// which has a max length. Domain name is used to make the names on BEB unique.
+type bebSubscriptionNameMapper struct {
+	domainName string
+	maxLength  int
+}
+
+func NewBebSubscriptionNameMapper(domainName string, maxNameLength int) *bebSubscriptionNameMapper {
+	return &bebSubscriptionNameMapper{
+		domainName: strings.TrimSpace(domainName),
```

Comment:
Why trimming space here? IMO it introduces more responsibility to this function.